src/SFML/Audio/SoundFileReaderWav.cpp

```diff
@@ -169,15 +169,29 @@ Uint64 SoundFileReaderWav::read(Int16* samples, Uint64 maxCount)
                 break;
             }
 
+            case 3:
+            {
+                Uint32 sample = 0;
+                if (decode24bit(*m_stream, sample))
+                    *samples++ = sample >> 8;
+                else
+                    return count;
+                break;
+            }
+
             case 4:
             {
-                Int32 sample = 0;
+                Uint32 sample = 0;
                 if (decode(*m_stream, sample))
                     *samples++ = sample >> 16;
                 else
                     return count;
                 break;
             }
+
+            default:
+                err() << "Failed to read from WAV sound file (unsupported sample size: " << m_bytesPerSample * 8 << " bit)" << std::endl;
+                return 0;
         }
```

Comment:
The check for unsupported sample size happens in the [`parseHeader()` method](https://github.com/Foaly/SFML/commit/101a09b12a388527f54d8fc51b76d0a59a731b51#diff-7d5da91722b151bf7289e00329995877R267). Also this behaviour is consitent with the [Flac implementation](https://github.com/SFML/SFML/blob/master/src/SFML/Audio/SoundFileReaderFlac.cpp#L141).